prime-dotnet-webapi/Controllers/OrganizationsController.cs

```diff
@@ -110,6 +123,116 @@ public async Task<ActionResult<Organization>> CreateOrganization(CreateOrganizat
             );
         }
 
+        // POST: api/Organizations/claim
+        /// <summary>
+        /// Claim an existing Organization.
+        /// </summary>
+        [HttpPost("claim", Name = nameof(ClaimOrganization))]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ApiResultResponse<int>), StatusCodes.Status200OK)]
+
+        public async Task<ActionResult<Organization>> ClaimOrganization(ClaimOrganizationViewModel claimOrganization)
+        {
+            if (!await _partyService.PartyExistsAsync(claimOrganization.PartyId, PartyType.SigningAuthority))
+            {
+                return BadRequest("Could not claim an organization, the passed in SigningAuthority does not exist.");
+            }
+
+            var organization = await _organizationService.ClaimOrganizationAsync(claimOrganization.PartyId, claimOrganization.PEC, claimOrganization.ClaimDetail);
+
+            return Ok(organization);
+        }
+
+        // GET: api/Organizations/claim
+        /// <summary>
+        /// Gets organization claim count by a given search criteria.
+        /// </summary>
+        [HttpGet("claim", Name = nameof(GetOrganizationClaimCount))]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ApiResultResponse<OrganizationClaim>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<int>> GetOrganizationClaimCount([FromQuery] OrganizationClaimSearchOptions search)
+        {
+            // check if the organization exists with the given PEC
+            if (!string.IsNullOrEmpty(search.Pec))
+            {
+                var organization = await _organizationService.GetOrganizationByPecAsync(search.Pec);
```

Comment:
This is fetching the entire Organization then throwing it away to see if it exists. Not a huge deal, but that's a bit wasteful. It feels like the `GetOrganizationClaimCountAsync` method could be written to return null if the org does not exist.

Reply:
The FE expects a BadRequest to be caught so it can't proceed to next step to claim the organization. If returning null, it means the ClaimOrganization does not exist instead. and the user can proceed. The alternative approach is to return a composite object containing both ClaimOrganizationExists & OrganizationExists which can be checked by the FE.

> This is fetching the entire Organization then throwing it away to see if it exists. Not a huge deal, but that's a bit wasteful. It feels like the `GetOrganizationClaimCountAsync` method could be written to return null if the org does not exist.